### ISSUE TITLE
Assertion `t >= 0 && t < n_classes` failed.

### ISSUE THREAD

**Issue by xings-sdnu** [YuvalNirkin/hyperseg#26]
Sorry to bother you, I have encontered a probelm: Assertion `t >= 0 && t < n_classes` failed when I try to use your model to my datasets, I have changed the num_class in my datasets, but I still face the problem, can u help me? here is my change:
self.classes = list(range(2))
self.image_classes = calc_classes_per_image(self.masks, 2, cache_file)

**Comment by YuvalNirkin** [comment]
Did you also specify `num_classes` in the arguments for the model?

**Comment by qjl988** [comment]
Hi，Did you end up getting that fixed?

**Comment by shengxing-phd** [comment]
Yes, it's the label's number in the dataset, I have changed the number, it works!&nbsp;


盛星
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "YuvalNirkin/hyperseg"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月24日(星期天) 下午2:59
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [YuvalNirkin/hyperseg] Assertion `t &gt;= 0 &amp;&amp; t < n_classes` failed. (Issue #26)





 
Hi，Did you end up getting that fixed?
